src/Api/Response/Response.php

```diff
@@ -90,4 +101,102 @@ public function getResultsCount()
+            ? 'did-you-mean' : $this->data[self::DATA_QUERY_INFO_MESSAGE]['queryStringType'];
+        $alternativeQuery = !empty($this->data[self::DATA_QUERY_INFO_MESSAGE]['didYouMeanQuery'])
+            ? $this->data[self::DATA_QUERY_INFO_MESSAGE]['didYouMeanQuery']
+            : $this->data[self::DATA_QUERY_INFO_MESSAGE]['currentQuery'];
+        $originalQuery = !empty($this->data[self::DATA_QUERY_INFO_MESSAGE]['didYouMeanQuery'])
+            ? $this->data[self::DATA_QUERY_INFO_MESSAGE]['currentQuery']
+            : $this->data[self::DATA_QUERY_INFO_MESSAGE]['originalQuery'];
+
+        if ($type === 'corrected') {
+            return $this->translator->trans(
+                'Findologic::Template.correctedQuery',
+                [
+                    'originalQuery' => $originalQuery,
+                    'alternativeQuery' => $alternativeQuery
+                ]
+            );
+        } else if ($type === 'improved') {
+            return $this->translator->trans(
+                'Findologic::Template.improvedQuery',
+                [
+                    'originalQuery' => $originalQuery,
+                    'alternativeQuery' => $alternativeQuery
+                ]
+            );
+        } else if ($this->data[self::DATA_QUERY_INFO_MESSAGE]['currentQuery']
+            && !empty($this->data[self::DATA_QUERY_INFO_MESSAGE]['currentQuery'])) {
+            return $this->translator->trans(
+                'Findologic::Template.queryInfoMessageQuery',
+                [
+                    'query' => $this->data[self::DATA_QUERY_INFO_MESSAGE]['currentQuery'],
+                    'hits' => $this->getResultsCount()
+                ]
+            );
+        } else if ($this->data[self::DATA_QUERY_INFO_MESSAGE]['selectedCategoryName']) {
+            return $this->translator->trans(
+                'Findologic::Template.queryInfoMessageCat',
+                [
+                    'filterName' => $this->getCategoryFilterName(),
+                    'cat' => $this->data[self::DATA_QUERY_INFO_MESSAGE]['selectedCategoryName'],
+                    'hits' => $this->getResultsCount()
+                ]
+            );
+        } else if ($this->data[self::DATA_QUERY_INFO_MESSAGE]['selectedVendorName']) {
+            return $this->translator->trans(
+                'Findologic::Template.queryInfoMessageVendor',
+                [
+                    'filterName' => $this->getVendorFilterName(),
+                    'vendor' => $this->data[self::DATA_QUERY_INFO_MESSAGE]['selectedVendorName'],
+                    'hits' => $this->getResultsCount()
+                ]
+            );
+        } else {
+            return $this->translator->trans(
+                'Findologic::Template.queryInfoMessageDefault',
+                [
+                    'hits' => $this->getResultsCount()
+                ]
+            );
+        }
+    }
+
+    private function getCategoryFilterName(): string
```

Comment:
I do not think we need these methods (`getCategoryFilterName`, `getVendorFilterName`) anymore.

Reply:
Filter name can't be retrieved from the request. In the request there is filter codes and values, but not Display name. 
For that reason, I am parsing the response for the Display name.

Reply:
You are totally right.